### PR TITLE
Task/04 styling

### PR DESCRIPTION
**Pull request by lesiaturuta**
Acceptance Criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Added layout of Search Bar
- [x]  Rewritten all styles with Module Styles
- [x]  Implemented dynamic theme change logic

Acceptance Criteria for 1*

- [x]  Implemented logic for fetching user's theme.
- [x]  Saved theme to the Local storage.
- [x] Note: if we have saved theme in Local storage, we use it first after reloading of the page

link:  https://lesiaturuta.github.io/fe-react-2024/

### CODE REVIEW

File: src/components/about/About.component.tsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import styles from './about.module.css';
 
-function App() {
+const App = ({ theme }: { theme: string }) => {
```
Comment (vadym-zinchenko-moc) [inline src/components/about/About.component.tsx]: why name `About` component is `App`?

File: src/components/about/About.component.tsx [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
     const name = 'Lesia';
-
     return (
-        <main className={styles.main}>
+        <main className={`${styles.main} ${theme === 'light' ? styles.light_theme : styles.dark_theme}`}>
```
Comment (vadym-zinchenko-moc) [inline src/components/about/About.component.tsx]: you can use `clsx` if you want for logic like this

File: src/components/footer/Footer.component.tsx [hunk 3]
```diff
@@ -12,8 +12,8 @@ const socialMedia = [
     { icon: linkedin, url: URL_LINK_IN },
 ];
 
-const Footer = () => (
-    <footer className={styles.footer}>
+const Footer = ({ theme }: { theme: string }) => (
+    <footer className={`${styles.footer} ${theme === 'light' ? styles.light_theme : styles.dark_theme}`}>
```
Comment (vadym-zinchenko-moc) [inline src/components/footer/Footer.component.tsx]: the same, I think you can use `clsx`

File: src/components/about/about.module.css [hunk 4]
```diff
@@ -1,10 +1,22 @@
+.light_theme {
```
Comment (vadym-zinchenko-moc) [inline src/components/about/about.module.css]: duplicating classes in each component is a bad idea. Why are you doing this? In the lecture, I explained how we can use variables. You need to change the approach. Declare the variables in the indeч.css file and then use them throughout the project

https://css-tricks.com/easy-dark-mode-and-multiple-color-themes-in-react/

File: src/components/footer/footer.module.css [hunk 5]
```diff
@@ -1,9 +1,21 @@
+.light_theme {
```
Comment (vadym-zinchenko-moc) [inline src/components/footer/footer.module.css]: the same, need to optimize

File: src/components/search/search.module.css [hunk 6]
```diff
@@ -0,0 +1,189 @@
+.light_theme {
```
Comment (vadym-zinchenko-moc) [inline src/components/search/search.module.css]: the same

File: src/components/themes/Themes.component.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import IconSun from '@/assets/icons/IconSun.component.tsx';
+import stylesHeader from '@/components/header/header.module.css';
+
+import styles from './themes.module.css';
+
+const ThemesComponent = ({ theme, changeTheme }: { theme: string; changeTheme: (name: string) => void }) => (
+    <div className="main">
+        <div className={stylesHeader.themes}>
+            <button onClick={() => changeTheme('light')} className={styles.theme_btn}>
+                <IconSun color={theme === 'light' ? '#fff' : undefined} />
```
Comment (vadym-zinchenko-moc) [inline src/components/themes/Themes.component.tsx]: it's bad to write `undefined`. You need use empty string or anther color

File: src/components/themes/Themes.component.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+            </button>
+            <div className={stylesHeader.verticalLine}></div>
+            <button onClick={() => changeTheme('dark')} className={styles.theme_btn}>
+                <IconMoon color={theme === 'dark' ? '#fff' : undefined} />
```
Comment (vadym-zinchenko-moc) [inline src/components/themes/Themes.component.tsx]: same

File: src/components/search/search.module.css [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+.drop_down_content .select_btn {
+  width: 100%;
+  height: 40px;
+  border-radius: 0;
+  background: var(--bg-input);
+}
+
+.drop_down_content .select_btn:hover {
+  background: #EF4934;
```
Comment (vadym-zinchenko-moc) [inline src/components/search/search.module.css]: you can use variable for this. Create in index.css and use in all project

File: src/components/search/search.module.css [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+  color: #FFF;
+}
+
+.select_btn {
+  color: var(--color-pimary);
+}
+
+.drop_btn {
+  background-color: #4CAF50;
+  color: #FFF;
```
Comment (vadym-zinchenko-moc) [inline src/components/search/search.module.css]: you can use variable for this. Create in index.css and use in all project

File: src/components/search/search.module.css [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+.sort_name {
+  font-size: 14px;
+  font-weight: 600;
+  line-height: 17px;
+  color: var(--color-pimary);
+  margin-right: 10px;
+}
+
+.select:focus {
+  background: #EF4934;
```
Comment (vadym-zinchenko-moc) [inline src/components/search/search.module.css]: you can use variable for this. Create in index.css and use in all project

File: src/components/search/search.module.css [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+  color: #FFF;
```
Comment (vadym-zinchenko-moc) [inline src/components/search/search.module.css]: same in all project (you can use variable for this. Create in index.css and use in all project)

File: src/components/productList/ProductList.component.tsx [hunk 13]
```diff
@@ -7,16 +8,29 @@ const ProductList = ({
...
     increaseCounter: () => void;
     decrementCounter: () => void;
+    theme: string;
 }) => (
-    <ul className={styles.products}>
-        {products.map((product: Product) => (
-            <ProductCard product={product} increaseCounter={increaseCounter} decrementCounter={decrementCounter} key={product.id} />
-        ))}
-    </ul>
+    <div className={`${styles.main} ${theme === 'light' ? styles.light_theme : styles.dark_theme}`}>
```
Comment (vadym-zinchenko-moc) [inline src/components/productList/ProductList.component.tsx]: you can use `clsx`

File: src/components/productCard/ProductCard.component.tsx [hunk 14]
```diff
@@ -44,7 +46,7 @@ const ProductCard = ({
                     </div>
 
                     <div className={styles.cart} onClick={changeCart}>
-                        <img className={styles.cart_button} src={cart} alt="cart" />
+                        <IconCart color={theme === 'light' ? undefined : '#FFF'} />
```
Comment (vadym-zinchenko-moc) [inline src/components/productCard/ProductCard.component.tsx]: it's bad to write undefined. You need use empty string or anther color

File: src/components/header/Header.component.tsx [hunk 15]
```diff
@@ -51,7 +56,7 @@ const Header = ({ amountCart, toggleCurrentPage }: { amountCart: number; toggleC
             </nav>
             <div className={styles.buttons}>
                 <div className={styles.cart}>
-                    <img className={styles.cart_button} src={cart} alt="cart" />
+                    <IconCart color={'#FFF'} />
```
Comment (vadym-zinchenko-moc) [inline src/components/header/Header.component.tsx]: what the reason to use props `color`. It's white by default

File: src/components/search/Search.component.tsx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+                    <button className={`${styles.btn} ${styles.mr_10}`}>Shoes</button>
+                    <button className={styles.btn}>Clothes</button>
+                </div>
+                <div className={styles.sort}>
+                    <span className={styles.sort_name}>Sort by:</span>
+
+                    <div className={styles.drop_down}>
+                        <button className={styles.select}>{selectedSort}</button>
+                        <div className={styles.drop_down_content}>
+                            {selectedSort !== nameSelect[0] && (
```
Comment (vadym-zinchenko-moc) [inline src/components/search/Search.component.tsx]: it's really bad approach for drop-dowm. Rewrite it pls.
https://www.w3schools.com/howto/howto_custom_select.asp
or use list <ul <li with loop.

File: src/components/search/Search.component.tsx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+import search from '@/assets/icons/search.svg';
+
+import styles from './search.module.css';
+
+const Search = ({ theme }: { theme: string }) => {
+    const nameSelect: string[] = ['Price (High - Low)', 'Price (Low- High)', 'Newest', 'Oldest'];
+
+    const [selectedSort, setSelectedSort] = React.useState<string>(nameSelect[0]);
+    return (
+        <div className={`${styles.main} ${theme === 'light' ? styles.light_theme : styles.dark_theme}`}>
```
Comment (vadym-zinchenko-moc) [inline src/components/search/Search.component.tsx]: maybe clsx?

File: src/components/productCard/ProductCard.component.tsx [hunk 18]
```diff
@@ -31,7 +33,7 @@ const ProductCard = ({
     };
 
     return (
-        <li className={styles.product}>
+        <li className={`${styles.product} ${theme === 'light' ? 'active' : ''}`}>
```
Comment (vadym-zinchenko-moc) [inline src/components/productCard/ProductCard.component.tsx]: maybe clsx?

File: src/components/search/search.module.css [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+.drop_down:hover .select {
+  background: var(--bg-input);
+  border-radius: 20px 20px 0 0;
+  border-bottom: none;
+}
+
+.drop_down:hover .select:hover {
+  background: #EF4934;
```
Comment (vadym-zinchenko-moc) [inline src/components/search/search.module.css]: check all colors in the whole project. And use variables

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/components/search/Search.component.tsx (hunk 16 above)
Comment (vadym-zinchenko-moc) [reply src/components/search/Search.component.tsx]: Screenshot 2024-05-21 at 17 17 20

and check colors pls

**Comment by vadym-zinchenko-moc** [comment]
1.
Screenshot 2024-05-21 at 17 17 42

2.
Screenshot 2024-05-21 at 17 18 48

3.
Screenshot 2024-05-21 at 17 21 08


check pls styles (seems like extra blue color) and line color must be another

File: src/components/about/About.component.tsx (hunk 1 above)
Comment (lesiaturuta) [reply src/components/about/About.component.tsx]: fixed

File: src/components/about/About.component.tsx (hunk 2 above)
Comment (lesiaturuta) [reply src/components/about/About.component.tsx]: added clsx

File: src/components/footer/Footer.component.tsx (hunk 3 above)
Comment (lesiaturuta) [reply src/components/footer/Footer.component.tsx]: added

File: src/components/about/about.module.css (hunk 4 above)
Comment (lesiaturuta) [reply src/components/about/about.module.css]: fixed

File: src/components/search/search.module.css (hunk 6 above)
Comment (lesiaturuta) [reply src/components/search/search.module.css]: fixed

File: src/components/themes/Themes.component.tsx (hunk 7 above)
Comment (lesiaturuta) [reply src/components/themes/Themes.component.tsx]: fixed

File: src/components/footer/footer.module.css (hunk 5 above)
Comment (lesiaturuta) [reply src/components/footer/footer.module.css]: fixed

File: src/components/themes/Themes.component.tsx (hunk 8 above)
Comment (lesiaturuta) [reply src/components/themes/Themes.component.tsx]: fixed

File: src/components/productList/ProductList.component.tsx (hunk 13 above)
Comment (lesiaturuta) [reply src/components/productList/ProductList.component.tsx]: fixed

File: src/components/productCard/ProductCard.component.tsx (hunk 18 above)
Comment (lesiaturuta) [reply src/components/productCard/ProductCard.component.tsx]: fixed

File: src/components/search/Search.component.tsx (hunk 17 above)
Comment (lesiaturuta) [reply src/components/search/Search.component.tsx]: fixed

File: src/components/productCard/ProductCard.component.tsx (hunk 14 above)
Comment (lesiaturuta) [reply src/components/productCard/ProductCard.component.tsx]: fixed

File: src/components/header/Header.component.tsx (hunk 15 above)
Comment (lesiaturuta) [reply src/components/header/Header.component.tsx]: fixed

File: src/components/search/search.module.css (hunk 19 above)
Comment (lesiaturuta) [reply src/components/search/search.module.css]: fixed

File: src/components/search/search.module.css (hunk 9 above)
Comment (lesiaturuta) [reply src/components/search/search.module.css]: fixed

File: src/components/search/search.module.css (hunk 10 above)
Comment (lesiaturuta) [reply src/components/search/search.module.css]: fixed

File: src/components/search/search.module.css (hunk 11 above)
Comment (lesiaturuta) [reply src/components/search/search.module.css]: fixed

File: src/components/search/search.module.css (hunk 12 above)
Comment (lesiaturuta) [reply src/components/search/search.module.css]: fixed

File: src/components/search/Search.component.tsx (hunk 16 above)
Comment (lesiaturuta) [reply src/components/search/Search.component.tsx]: fixed

File: src/components/about/About.component.tsx [hunk 20]
```diff
@@ -1,33 +1,40 @@
...
     const name = 'Lesia';
-
     return (
-        <main className={styles.main}>
+        <main
+            className={clsx({
+                [styles.main]: true,
+                light_theme: theme === 'light',
+                dark_theme: theme === 'dark',
+            })}
```
Comment (vadym-zinchenko-moc) [inline src/components/about/About.component.tsx:12]: `clsx(styles.main, {
    light_theme: theme === 'light',
    dark_theme: theme === 'dark',
})`

can we use like this?

File: src/components/footer/Footer.component.tsx [hunk 21]
```diff
@@ -12,8 +14,14 @@ const socialMedia = [
     { icon: linkedin, url: URL_LINK_IN },
 ];
 
-const Footer = () => (
-    <footer className={styles.footer}>
+const Footer = ({ theme }: { theme: string }) => (
+    <footer
+        className={clsx({
+            [styles.footer]: true,
```
Comment (vadym-zinchenko-moc) [inline src/components/footer/Footer.component.tsx]: same

File: src/components/search/Search.component.tsx [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+    const nameSelect: string[] = ['Price (High - Low)', 'Price (Low- High)', 'Newest', 'Oldest'];
+
+    const [selectedSort, setSelectedSort] = React.useState<string>(nameSelect[0]);
+    const [isOpenDropDropDown, setIsOpenDropDropDown] = React.useState<boolean>(false);
+
+    // , setSelectedSort] = React.useState<string>(nameSelect[0]);
+    return (
+        <div
+            className={clsx({
+                [styles.main]: true,
```
Comment (vadym-zinchenko-moc) [inline src/components/search/Search.component.tsx]: same

File: src/components/search/Search.component.tsx [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+                light_theme: theme === 'light',
+                dark_theme: theme === 'dark',
+            })}
+        >
+            <div className={styles.search}>
```
Comment (vadym-zinchenko-moc) [inline src/components/search/Search.component.tsx:23]: Screenshot 2024-05-23 at 13 23 38

the search bar in the mobile version does not match the design

File: src/components/search/Search.component.tsx [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+                    <button className={clsx(styles.btn, styles.mr_10)}>Shoes</button>
+                    <button className={styles.btn}>Clothes</button>
+                </div>
+                <div className={styles.sort}>
+                    <span className={styles.sort_name}>Sort by:</span>
+
+                    <div
+                        className={clsx({
+                            [styles.drop_down]: true,
+                            // [styles.select_background]: isOpenDropDropDown,
```
Comment (vadym-zinchenko-moc) [inline src/components/search/Search.component.tsx]: same, and delete extra code

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/components/about/About.component.tsx (hunk 20 above)
Comment (lesiaturuta) [reply src/components/about/About.component.tsx:12]: fixed

File: src/components/search/Search.component.tsx (hunk 22 above)
Comment (lesiaturuta) [reply src/components/search/Search.component.tsx]: fixed

File: src/components/footer/Footer.component.tsx (hunk 21 above)
Comment (lesiaturuta) [reply src/components/footer/Footer.component.tsx]: fixed

File: src/components/search/Search.component.tsx (hunk 23 above)
Comment (lesiaturuta) [reply src/components/search/Search.component.tsx:23]: fixed

File: src/components/search/Search.component.tsx (hunk 24 above)
Comment (lesiaturuta) [reply src/components/search/Search.component.tsx]: fixed

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**